types/instance.d.ts

```diff
@@ -38,7 +38,7 @@ export interface FastifyInstance<
   decorateRequest(property: string | symbol, value: any, dependencies?: string[]): FastifyInstance<RawServer, RawRequest, RawReply, Logger>;
   decorateReply(property: string | symbol, value: any, dependencies?: string[]): FastifyInstance<RawServer, RawRequest, RawReply, Logger>;
 
-  hasDecorator(decorator: string | symbol): boolean;
+  hasDecorator<K extends string | symbol, T = any>(decorator: K): this is FastifyInstance<RawServer, RawRequest, RawReply, Logger> & Record<K, T>;
```

Comment:
I would like to swap generics to make possible pass only decorator type: `<T = any, K extends string = string>`

Reply:
Would be ideal, and I also first tried that, but TypeScript complains if you put optional parameters after required parameters.